src/pdf_parser/pdf_utils/disambiguator/tests/conftest.py

```diff
@@ -0,0 +1,175 @@
+import pytest
+from layoutparser import TextBlock, Rectangle, Layout
+import random
+
+
+def generate_text_block(coordinates: tuple, score: float) -> TextBlock:
+    """Generates a textblock with specific coordinates and a random score."""
+    return TextBlock(
+        block=Rectangle(
+            x_1=coordinates[0],
+            y_1=coordinates[1],
+            x_2=coordinates[2],
+            y_2=coordinates[3],
+        ),
+        text=None,
+        id=None,
+        type="Title",
+        parent=None,
+        next=None,
+        score=score,
+    )
+
+
+@pytest.fixture(scope="function")
+def test_layout_random(request) -> Layout:
+    """Returns a layout with the given number of text blocks."""
+    return Layout(
+        [
+            generate_text_block(
+                coordinates=(
+                    random.random() * 100,
+                    random.random() * 100,
+                    random.random() * 100,
+                    random.random() * 100,
+                ),
```

Comment:
Nope. This will give weird coordinates that are not consistent with (x_top_left, y_top_left, x_bottom_right, y_bottom_right). Obviously many ways of doing this, something like:

# Set the range of the box coordinates
x_min, x_max = 0, 100
y_min, y_max = 0, 100
num_boxes=100

# Generate the random coordinates for all boxes
x1 = np.random.uniform(x_min, x_max, size=num_boxes)
y1 = np.random.uniform(y_min, y_max, size=num_boxes)
x2 = np.random.uniform(x1, x_max, size=num_boxes)
y2 = np.random.uniform(y1, y_max, size=num_boxes)

# Sort the coordinates so that x1 < x2 and y1 < y2
box_coords = np.column_stack((np.sort([x1, x2], axis=0), np.sort([y1, y2], axis=0))).T

Also

Reply:
Set the seed too for repeatability

Reply:
I've updated the code to make sure x1 < x2 and y1 < y2 if you want to check.